GoogleForCreators/web-stories-wp
# Dashboard: support filtering stories by taxonomy

## Context


Add programmatic taxonomy filters to the dashboard.

## Summary


Give the user the ability to filter by story taxonomies.  Similar to the author filter.

## Relevant Technical Choices


There can be multiple taxonomy filters, and a programmatic solution is needed to build out these filters.  The solution used here lends itself to encompass all dashboard filters state into one place making the filtering process simpler later on.

## To-do


Move all dashboard story filters into the new provider to simplify dashboard story filtering

## User-facing changes


before/after
https://user-images.githubusercontent.com/14057928/171685426-33d5664b-0485-4567-8773-d45d6aae5bf8.mp4


## Testing Instructions






This PR can be tested by following these steps:

1. Go to story dashboard

On Staging:
- Notice there are no taxonomy filters

On Branch:
- Notice there are taxonomy filters (categories, verticals)
2. Make sure there are stories with taxonomies assigned to them
3. Test filters correctly filters stories
4. Test that the search within the filter dropdown is working correctly as well


## Reviews

### Does this PR have a security-related impact?
No


### Does this PR change what data or activity we track or use?
No


### Does this PR have a legal-related impact?
No


## Checklist




---



Fixes #11436
